sources/engine/Stride.Input/Windows/InputSourceWindowsDirectInput.cs

```diff
@@ -116,6 +118,43 @@ public void OpenDevice(DeviceInstance deviceInstance)
                 RegisterDevice(controller);
             }
         }
+
+        private bool IsXInputDevice(Guid productGuid)
+        {
+            // Set security level to IMPERSONATE
+
+            DComSessionOptions DComOptions = new DComSessionOptions();
+            DComOptions.Impersonation = ImpersonationType.Impersonate;
+
+            var mySession = CimSession.Create(null, DComOptions);
+
+            // Select all device IDs that contain "IG_".  If so, it's an XInput device
+            // This information can not be found from DirectInput 
+            IEnumerable<CimInstance> allDevices = mySession.QueryInstances(@"root\cimv2", "WQL", "SELECT DeviceID FROM Win32_PNPEntity WHERE DeviceId LIKE '&IG_%'");
```

Comment:
Are you sure this is optimization was correct?
`LIKE '&IG_%'` would match strings that start with `&IG_`, but in the regex you previously had that at the end of the string.

Reply:
Oh, yea my bad little typo, I fixed now